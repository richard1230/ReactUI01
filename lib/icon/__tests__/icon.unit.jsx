import * as render from 'react-test-renderer'
import React from "react";
import Icon from "../icon";
import Enzyme, {mount} from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';


Enzyme.configure({adapter: new Adapter()});

describe('icon 组件单元测试: ', () => {
  it('should render successful', function () {
    const json = render.create(<Icon name='alipay'/>).toJSON()
    expect(json).toMatchSnapshot()
  });
  it('onClick', () => {
    const fn = jest.fn()
    const component = mount(<Icon name="alipay" onClick={fn}/>)
    component.find('svg').simulate('click')
    expect(fn).toBeCalled()
  });

})