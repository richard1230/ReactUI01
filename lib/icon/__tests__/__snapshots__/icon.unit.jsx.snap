// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon 组件单元测试:  should render successful 1`] = `
<svg
  className="fui-icon"
>
  <use
    xlinkHref="#alipay"
  />
</svg>
`;
